crates/bevy_ecs/src/archetype.rs

```diff
@@ -23,14 +23,30 @@ use crate::{
     bundle::BundleId,
     component::{ComponentId, StorageType},
     entity::{Entity, EntityLocation},
-    storage::{ImmutableSparseSet, SparseArray, SparseSet, SparseSetIndex, TableId},
+    storage::{ImmutableSparseSet, SparseArray, SparseSet, SparseSetIndex, TableId, TableRow},
 };
 use std::{
     collections::HashMap,
     hash::Hash,
     ops::{Index, IndexMut},
 };
 
+/// An opaque location within a [`Archetype`].
+///
+/// This can be used in conjunction with [`ArchetypeId`] to find the exact location
+/// of an [`Entity`] within a [`World`]. An entity's archetype and index can be
+/// retrieved via [`Entities::get`].
+///
+/// [`World`]: crate::world::World
+/// [`Entities::get`]: crate::entity::Entities
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[repr(transparent)]
+pub struct ArchetypeIndex(usize);
+
+impl ArchetypeIndex {
+    pub(crate) const INVALID: ArchetypeIndex = ArchetypeIndex(usize::MAX);
```

Comment:
```suggestion
    pub const INVALID: ArchetypeIndex = ArchetypeIndex(usize::MAX);
```
Seems reasonable to make this public :thinking: Is there a reason we dont have public `ArchetypeRow::new` and `ArchetypeRow::index` functions.

Reply:
Done.